lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js

```diff
@@ -363,6 +503,23 @@ describe('LMSFilePicker', () => {
         fileList.prop('noFilesMessage').props.href,
         'https://fake_help_link'
       );
+      // After first file fetch, we're at the top level, so the no-files message
+      // should operate in its "course" context
+      assert.isFalse(fileList.prop('noFilesMessage').props.inSubfolder);
+    });
+
+    it('renders no-file message in folder context if in a subfolder', async () => {
+      const wrapper = renderFilePicker({ withBreadcrumbs: true });
+      await fakeApiCall;
+      wrapper.update();
+      await act(() => changePath(wrapper, fakeFolders[0]));
+      await fakeApiCall;
+      wrapper.update();
```

Comment:
The `waitForElement` helper will update the wrapper, so you don't need to do that.